Adding a specific hash methods to SafetyUtil.

hchoura opened the pull request:
The main reason to avoid converting object.

hchoura committed:
Adding a specific hash methods to SafetyUtil.

The main reason to avoid converting object.

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+    return java.util.Objects.hash(o1);
+  }
+
+  public static int hash(FObject o1) {
```
Is this correct? Don't we just want to call the hashCode() method on the object that we generated?

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hash(boolean o1) {
+    return java.util.Objects.hash(o1);
```
This will convert it to an Object, which is what we were trying to avoid. Something like this:
return o1 ? 1 : 0;

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hash(String o1) {
+    return java.util.Objects.hash(o1);
```
Just return o1.hashCode(); or to be safe:
return o1 == null ? 0 : o1.hashCode();

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hash(short o1) {
+    return java.util.Objects.hash(o1);
```
return o1;

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hash(float o1) {
+    return java.util.Objects.hash(o1);
```
Something like
return Float.toIntBits(o1);

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hash(double o1) {
+    return java.util.Objects.hash(o1);
```
Use Double.doubleToRawLongBits(double value)

hchoura committed:
change generated method from hash to hashcode

make hashCode more optimal.

Commit:
add condition if user == null will return empty string

kgrgreer committed:
Fix collider to not remove children while colliding to avoid corruption and exceptions.

noodlemoodle committed:
Update src/foam/nanos/approval/ApprovableAware.js

Co-Authored-By: Scott Head <43998323+nanoscott@users.noreply.github.com>

kgrgreer committed:
Merge pull request #3390 from yij793/NP-226/add-agent-to-history-record

add agent to history record to speicific the agent and business

jinnnn9 committed:
refactor change password view

jinnnn9 committed:
remove trailing comma

jinnnn9 committed:
typo fix

jinnnn9 committed:
addresses comments

kgrgreer committed:
Merge pull request #3386 from noodlemoodle/fixNoDupeARsForUPDATE

fixed noDuplicateApprovalRequests logic for Update

kgrgreer committed:
Fix Enum.asJavaValue support.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

jinnnn9 committed:
address comments on pr 2

kgrgreer committed:
Fix Collider bugs.

kgrgreer committed:
Initial check-in of spacewar demo/game.

jinnnn9 committed:
address comment 2

nanopay-arthur committed:
allow ChoiceView to take in async label functions in objToChoice

nanopay-arthur committed:
switch out onDAOUpdate callback functions to arrow functions

jlhughes committed:
Merge pull request #3397 from nanopay-arthur/NP-869/userNotDisplayedInTicketsBug

[NP-869] - Give ChoiceView Ability To Take in Async Label Functions In objToChoice

Commit:
Added approval request cancellation functionality

jinnnn9 committed:
remove trailing commas

jinnnn9 committed:
revert SectionView back to previous version + add css property for button in ChangePasswordView

jinnnn9 committed:
remove trailing quotation mark

hchoura committed:
change the condition & fix the boolean variable

kgrgreer committed:
Work on SpaceWar game.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
SpaceWar updates.

kgrgreer committed:
Update SpaceWar ship design.

minsunny committed:
added a label on deleteModal

jinnnn9 committed:
enable ChangePasswordView to decide if fields should be displayed vertically or horizontally

jinnnn9 committed:
documentation fix

nanopay-arthur committed:
set ticket owner relationship visibility to RO

kgrgreer committed:
Update comments.

kgrgreer committed:
Fix JS test running which wasn't always updating error output or lastRunTime and was saving to DAO twice.

kgrgreer committed:
Merge pull request #3400 from nanopay-arthur/NP-869/userNotDisplayedInTicketsBug

[NP-869] - Set Ticket Owner Relationship Visibility To RO

jlhughes committed:
Merge pull request #3399 from foam-framework/CPF-1729/adding_deleteModal_sendRequestMoney

Added a label on deleteModal

noodlemoodle committed:
get user from dao instead of building

kgrgreer committed:
Merge pull request #3401 from noodlemoodle/liquidapprovalbug2

get user from dao instead of building

jinnnn9 committed:
address comments on pr 3

jinnnn9 committed:
add comment on subtitle css property

kgrgreer committed:
Merge pull request #3395 from jinn9/CPF-3313/refactorChangePasswordView

[CPF-3313] Refactor change password view

Commit:
Added proper fix

kgrgreer committed:
Update to SpaceWar game.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Fix copyright.

kgrgreer committed:
Update copyright.

kgrgreer committed:
Ported FOAM1 interpolation to FOAM2.

kgrgreer committed:
Add interpolation support to Animation.

kgrgreer committed:
Improve snake explosion using interpolation.

kgrgreer committed:
Improve Snake game.

kgrgreer committed:
Add import.

kgrgreer committed:
Minor improvements to SpaceWar game.

kgrgreer committed:
Add 'delay' ability to Animations.

kgrgreer committed:
Add ship explosion to SpaceWar.

kgrgreer committed:
Small fix.

kgrgreer committed:
Simplify Spacewar.

kgrgreer committed:
Minor Collider updates.

kgrgreer committed:
Include arcWidth for Circle bounds.

kgrgreer committed:
Fix SpaceWar to not add bullets twice.

jlhughes committed:
Merge pull request #3393 from zahnno/appconfig-refactor

Support phone to Appconfig & Refactor

nanopay-arthur committed:
add strategizerDisabled property to FObjectView

nanopay-arthur committed:
refactor strategizerDisabled property to enableStrategizer

kgrgreer committed:
Fix DAOtoFObjectArrayView to use Stack properly.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Merge pull request #3402 from nanopay-arthur/NP-972/addDisableStrategizerPropertyFObjectView

[NP-972] - Add strategizerDisabled Property To FObjectView

Commit:
Added a fulfillment to rejections and cancellations

kgrgreer committed:
Merge pull request #3403 from tharmaman/NP-820/UpdateRejectButtonToCancelFOAM

Adding a Cancellation Feature on Approval Request

noodlemoodle committed:
move Authorizable interface to foam/nanos/auth

noodlemoodle committed:
remove string id from approvableaware interface

kgrgreer committed:
Add Element.addClasses(), which was needed for the new Enum views.

kgrgreer committed:
Add support for colouring and styling enum values, with view support and a demo.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

jlhughes committed:
Merge pull request #3408 from noodlemoodle/removeStringId

Remove stringId from approvableaware

kgrgreer committed:
Rename enum bold property to avoid name conflict with ISO20022 support.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Fix enum changes breaking Java build.

nanoNeel committed:
skip permission check if the user is the file owner

kgrgreer committed:
Switch to ArrayParser singleton and fix Array.cast().

kgrgreer committed:
Formatting.

kgrgreer committed:
Remove trailing comma.

kgrgreer committed:
Formatting.

kgrgreer committed:
Add comment.

kgrgreer committed:
Add printing of exception stack-trace to help debugging.

kgrgreer committed:
Add new Projection mlang sink.

Commit:
Allow sectioneddetailview to define sections

Commit:
Update documentation

Commit:
Omit unused actions and props if useSections exists(sections)

noodlemoodle committed:
put diff hashmap into a treemap so the props can be iterated thru in order

jlhughes committed:
Support mapping Themes to URL domain.
ThemeDomainsDAO manages a reverse lookup of domains to themes.
Theme now includes AppConfig which is applied before Group AppConfig settings.

NP-65 #time 4h

jlhughes committed:
correct nspec setup and
add missing delegate.put in ThemeDomainsDAO

noodlemoodle committed:
do not initialize treemap if hashmap is null

jlhughes committed:
Merge pull request #3410 from nanoNeel/NP-1013/fix-file-update

skip permission check if the user is the file owner

noodlemoodle committed:
added a comment

kgrgreer committed:
Merge pull request #3414 from noodlemoodle/ruby/fix-for-approvablehashkey

put diff hashmap into a treemap so the props can be iterated thru in order

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into NP-65/Themes

lchanmann committed:
Add NewEqOld and Dot strategy references

jlhughes committed:
Remove appName, priority

lchanmann committed:
Fix data is undefined inside strategizer query and use strategy id as choice value

lchanmann committed:
Fix client-side ContextObject parsing

Without f() on javascript side the ContextObject would be parsed as Constant

jlhughes committed:
Fall through to domain theme if user personal theme not found.

jlhughes committed:
update tableCellFormatter arrow function

jlhughes committed:
Merge pull request #3416 from lchanmann/FixRulePredicateView

Fix meter rules predicate view

bshepherdson committed:
Add missing @license headers

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into NP-65/Themes

KolombetM committed:
Remove read permission check

jlhughes committed:
Correct comments around license header

jlhughes committed:
Merge pull request #3418 from foam-framework/foamlink-license-header

Correct comments around license header

kgrgreer committed:
Fix for Enum's not including the enum name in their CSS class.

jlhughes committed:
Merge branch 'NP-65/Themes' of github.com:foam-framework/foam2 into NP-65/Themes

kgrgreer committed:
Fix spacing.

kgrgreer committed:
Formatting.

kgrgreer committed:
Formatting.

kgrgreer committed:
Add virtual _choiceText_ property to NSpec.

jlhughes committed:
Merge pull request #3417 from Mykola97/NP-1060/Receiver-of-the-attachment

[NP-1060] Receiver of the attachment is not able to view it

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into NP-65/Themes

Commit:
Merge branch 'master' of github.com:foam-framework/foam2 into define-sections

nanopay-arthur committed:
remove label method from User

nanopay-arthur committed:
rename label method references to toSummary

jlhughes committed:
Merge pull request #3420 from nanopay-arthur/NP-1020/labelToToSummary

[NP-1020] Label to ToSummary

kgrgreer committed:
Remove ProxyDAO's dependency on NANOS logger.

Commit:
add white color to theme

kgrgreer committed:
Merge pull request #3421 from yij793/NP-53/add-white-color-to-theme

add white color to theme

kgrgreer committed:
Add swap() method to Collider.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

lchanmann committed:
Add checkerPredicate to ApprovableAware and support

lchanmann committed:
Remove approvableAwareEnabled from easyDAO

kgrgreer committed:
Prevent saying 'Game Over' multipile times in Snake game.

kgrgreer committed:
Formatting.

kgrgreer committed:
Formatting.

kgrgreer committed:
Add AllViews to nanos files.

nanopay-arthur committed:
give RichChoiceView searchBy support

lchanmann committed:
Deprecate approvableAwareEnabled

To be removed after services migration.

nanopay-arthur committed:
refactor filter_ property

hchoura committed:
re-throw the exception

hchoura committed:
re-throw the exception

hchoura committed:
re-throw the exception

hchoura committed:
Make the test Throwable

Commit:
Fixed approval requests to show pending

jlhughes committed:
Merge pull request #3428 from nick-prat/ApprovalRequestPendingFix

Fixed approval requests to show pending

jlhughes committed:
Merge pull request #3423 from nanopay-arthur/NP-843/giveRichChoiceViewSearchBySupport

[NP-843] - Give RichChoiceView searchBy Support

jlhughes committed:
Merge pull request #3422 from lchanmann/NP-1062/AddCheckerPredicateToApprovableAware

Add checkerPredicate to ApprovableAware

kgrgreer committed:
Remove stale references to BrowseView in menus.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Fix typo in CRUNCH rules.

Commit:
Use reduce for useSections, Added class prop to detail prop view

kgrgreer committed:
Merge pull request #3412 from zahnno/define-sections

Allow Sectioned Detail View to define sections

kgrgreer committed:
Minor updates to SpaceWar game.

kgrgreer committed:
Add temporary warning to detect Limit not working as expected.

kgrgreer committed:
Formatting.

kgrgreer committed:
Formatting.

kgrgreer committed:
Fix for multi-part key parsing when one part contains a timestamp.

kgrgreer committed:
Parser re-ordering in AnyParser.

kgrgreer committed:
Make DateParser stricter.

kgrgreer committed:
Make ExpParser also handle dates.

kgrgreer committed:
Formatting.

kgrgreer committed:
Make ObjectDateParser a Singleton. Add ability to parse long-form Dates also.

kgrgreer committed:
Add setOutputReadableDates() method to JSON Outputter.

kgrgreer committed:
Sort imports.

kgrgreer committed:
Formatting.

nanopay-arthur committed:
remove margin-left and refactor select-level width

kgrgreer committed:
Merge pull request #3430 from nanopay-arthur/CPF-4694/styleSideNavBarMenuLabels

[CPF-4694] - TreeView select-level Width Update

kgrgreer committed:
Unfinished PROJECTION support for ChoiceView.

kgrgreer committed:
Add limit to GROUP_BY.

kgrgreer committed:
Remove errors when attempting to create Impl classes from abstract classes.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

jlhughes committed:
Revert "Fixed approval requests to show pending"

jlhughes committed:
Merge pull request #3431 from foam-framework/revert-3428-ApprovalRequestPendingFix

Revert "Fixed approval requests to show pending"

Commit:
NP-998 Phone Number View Hide

Commit:
Format address grid columns aesthetically

Commit:
add theme to context

kgrgreer committed:
Merge pull request #3424 from foam-framework/throwException1

re-throw the exception

kgrgreer committed:
Merge pull request #3425 from foam-framework/throwException2

re-throw the exception

kgrgreer committed:
Merge pull request #3426 from foam-framework/throwException3

re-throw the exception

Commit:
Address Comments

kgrgreer committed:
Formatting.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Merge pull request #3427 from foam-framework/makeTestThrowable

Make the test Throwable

Commit:
Working on fixing detail view

Commit:
Detail view approval field shows correct value

nick-prat committed:
Merge branch 'master' into ApprovalRequestPendingFix

jlhughes committed:
Merge pull request #3436 from nick-prat/ApprovalRequestPendingFix

ApprovalRequest detail view approver fix

Commit:
using them.supportUser method to get phone and email

jlhughes committed:
Merge pull request #3435 from yij793/NP-65/addtion-update-for-theme

add theme to context

jlhughes committed:
Merge branch 'NP-65/Themes' of github.com:foam-framework/foam2 into NP-65/Themes

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into NP-65/Themes

jlhughes committed:
correct bad merge

olhabn committed:
fixing compare

kgrgreer committed:
Merge pull request #3365 from olhabn/NP-638/OrderBy_support_multiple_ordering_arguments

Fixing thenby

kgrgreer committed:
Remove stale references to BrowseView which doesn't exist.

jlhughes committed:
Merge pull request #3432 from carl-zzz/NP-998/PhoneNumber

NP-998 Phone Number View Hide

Commit:
add new theme() to context if theme is null

Commit:
resolve conflict

Commit:
sdd generic action

Commit:
replace supportUser in theme with support email and suppor Phone

Commit:
add more args to email template

Commit:
array change

Commit:
change to simple cast

Commit:
move condition to top

dependabot[bot] committed:
Bump Java-WebSocket from 1.3.4 to 1.5.0 in /tools

Bumps [Java-WebSocket](https://github.com/TooTallNate/Java-WebSocket) from 1.3.4 to 1.5.0.
- [Release notes](https://github.com/TooTallNate/Java-WebSocket/releases)
- [Changelog](https://github.com/TooTallNate/Java-WebSocket/blob/master/CHANGELOG.md)
- [Commits](https://github.com/TooTallNate/Java-WebSocket/compare/Java-WebSocket-1.3.4...v1.5.0)

Signed-off-by: dependabot[bot] <support@github.com>

kgrgreer committed:
Fix for GroupBy not working properly with CachingDAO.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

jlhughes committed:
Merge pull request #3438 from yij793/NP-65/sub-add-new-theme-if-theme-is-none

add new theme() to context if theme is null

kgrgreer committed:
Merge pull request #3440 from foam-framework/dependabot/maven/tools/org.java-websocket-Java-WebSocket-1.5.0

Bump Java-WebSocket from 1.3.4 to 1.5.0 in /tools

kgrgreer committed:
Merge pull request #3433 from zahnno/grid-column-address

Format address grid columns aesthetically

kgrgreer committed:
Merge pull request #3394 from foam-framework/UMLDiagramPermission2

Fix the UML diagram

kgrgreer committed:
Merge pull request #3378 from foam-framework/MakeLabelDynamically

Make BooleanFilterView label dynamically

kgrgreer committed:
Merge pull request #3377 from olhabn/NP-824/fixing_google_sheets_credentials

Np 824/fixing google sheets credentials

kgrgreer committed:
Merge pull request #3295 from foam-framework/lessCode

Refactoring Property Info

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hashCode(float o1) {
+    return Float.floatToIntBits(o1);
```
Something like
return Float.toIntBits(o1);

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
   }
 
+  public static int hashCode(Object o1) {
+    return java.util.Objects.hash(o1);
```
Faster to do:
return o1 == null ? 0 : o1.hashCode();

kgrgreer replied:
Objects.hash() creates an array which generates garbage and is slower.

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hashCode(FObject o1) {
+    return java.util.Objects.hash(o1);
```
Faster to do:
return o1 == null ? 0 : o1.hashCode();

kgrgreer replied:
Objects.hash() creates an array which generates garbage and is slower.

kgrgreer commented on src/foam/util/SafetyUtil.java:
```diff
@@ -100,6 +100,42 @@ public static int compare(double o1, double o2) {
+  }
+
+  public static int hashCode(long o1) {
+    return java.util.Objects.hashCode(o1);
```
Something like this would be faster:
(int)(o1()^(o1>>>32))

kgrgreer replied:
Not resolved!

jlhughes committed:
Another refactor so support Themes on Group
Also add Address to Theme

jlhughes committed:
Remove accidentally added file

jlhughes committed:
Merge pull request #3413 from foam-framework/NP-65/Themes

[NP-65] Support mapping Themes to URL domain.

kgrgreer committed:
Remove stale code.

kgrgreer committed:
Initial check-in of fire demo.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Fix buiild.

jlhughes committed:
Revert "[NP-65] Support mapping Themes to URL domain."

jlhughes committed:
Merge pull request #3441 from foam-framework/revert-3413-NP-65/Themes

Revert "[NP-65] Support mapping Themes to URL domain."

jlhughes committed:
Revert "Revert "[NP-65] Support mapping Themes to URL domain.""

jlhughes committed:
Add helper Model Themes

jlhughes committed:
Merge pull request #3442 from foam-framework/revert-3441-revert-3413-NP-65/Themes

Revert "Revert "[NP-65] Support mapping Themes to URL domain.""

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into NP-65/Themes

jlhughes committed:
Cleanup Theme logic with use of Themes

jlhughes committed:
When finding group theme, iterator through parents

jlhughes committed:
Only rethrow exception when mismatch from expected exception

jlhughes committed:
Merge pull request #3444 from foam-framework/NP-65/Themes

[NP-65] Themes Refactor

Commit:
change email arg link to appLink due to link is occupayed

kgrgreer committed:
Merge pull request #3445 from yij793/NP-948/change-args-link

change email arg link to appLink due to link is occupayed

Commit:
cleanup

jlhughes committed:
Overlay AppConfig with Theme.appConfig during configure()

jlhughes committed:
Configure arugment url is Request is null

jlhughes committed:
Merge pull request #3447 from foam-framework/NP-65/Themes

Overlay AppConfig with Theme.appConfig during configure()

Commit:
Fix ReadOnlyEnumView label

kgrgreer committed:
Merge pull request #3448 from KernelDeimos/NP-627/FixEnumViewLabel

[NP-627] Fix ReadOnlyEnumView label

Commit:
missing get.

nanoMichal committed:
Update src/foam/nanos/ruler/action/AbstractCheckDAOforMatching.js

nanoMichal committed:
Update src/foam/nanos/ruler/action/AbstractCheckDAOforMatching.js

nanoMichal committed:
Update src/foam/nanos/ruler/action/AbstractCheckDAOforMatching.js

kgrgreer committed:
Don't fire property change events for Date values that don't change.

kgrgreer committed:
Merge branch 'master' of https://github.com/foam-framework/foam2

kgrgreer committed:
Formatting.

hchoura committed:
add Context to sessionServerBox

hchoura committed:
Update src/foam/box/SessionServerBox.java

Co-authored-by: Joel Hughes <jlhughes@users.noreply.github.com>

jlhughes committed:
Merge pull request #3451 from foam-framework/addContextToSessionServerBox

add Context to sessionServerBox

jlhughes committed:
Correct Address package
and add Multiview to Image properties

jlhughes committed:
Merge pull request #3453 from foam-framework/NP-65/ThemeMultiView

Correct Address package

jlhughes committed:
Merge pull request #3439 from nanoMichal/genericAction

Add generic dao filtering action based on comparing object properties

hchoura committed:
Merge branch 'hashcode1' of git@github.com:foam-framework/foam2.git into hashcode1

hchoura committed:
Fix HashCode for Object and FObject
